### PR TITLE
fix(bot-template): click "Show profile" in bot app didn't response sometimes

### PR DESCRIPTION
**Pull request by SLdragon**
Fix this issue: https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_sprints/taskboard/AuthAndData/Microsoft%20Teams%20Extensibility/CY21-6.1?workitem=9997644

### CODE REVIEW

File: templates/bot/js/default/dialogs/mainDialog.js [hunk 1]
```diff
@@ -58,6 +59,16 @@ class MainDialog extends RootDialog {
     return await stepContext.beginDialog(TEAMS_SSO_PROMPT_ID);
   }
 
+  async dedupStep(stepContext) {
+    const tokenResponse = stepContext.result;
+    if (tokenResponse) {
```
Comment (blackchoey) [inline templates/bot/js/default/dialogs/mainDialog.js:64]: please add some comments to help users understand the logic

File: templates/bot/js/default/dialogs/mainDialog.js [hunk 2]
```diff
@@ -58,6 +59,16 @@ class MainDialog extends RootDialog {
...
+      if (await this.shouldDedup(stepContext.context)) {
+        return Dialog.EndOfTurn;
```
Comment (wenytang-ms) [inline templates/bot/js/default/dialogs/mainDialog.js:66]: if there need to add log?

File: templates/bot/js/default/dialogs/mainDialog.js (hunk 1 above)
Comment (SLdragon) [reply templates/bot/js/default/dialogs/mainDialog.js:64]: Sure

File: templates/bot/js/default/dialogs/mainDialog.js (hunk 2 above)
Comment (SLdragon) [reply templates/bot/js/default/dialogs/mainDialog.js:66]: OK, thanks

**Review by YitongFeng-git — COMMENTED** [review]
Please also change the templates msgext-bot ts and js, thanks!

**Review by wenytang-ms — APPROVED** [review]

**Review by blackchoey — APPROVED** [review]

**Pull request merged**